test/performance/range/view/view_drop_view_take_benchmark.cpp

```diff
@@ -0,0 +1,138 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <deque>
+#include <forward_list>
+#include <list>
+#include <string>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/drop.hpp>
+#include <seqan3/range/view/take.hpp>
+
+using namespace seqan3;
+
+// THIS FILE IMPLICITLY TESTS seqan3::view::slice, because that is just drop piped into take
+
+template <typename container_t, typename drop_t, typename take_t>
+void iterate_left_to_right(benchmark::State& state)
+{
+    container_t c;
+    c.resize(1'000'000);
+    uint8_t i = 0;
+    for (auto & e : c)
+        e = ++i; // dummy values
+
+    unsigned dummy = 0;
+
+    if constexpr (std::Same<drop_t, void>)
+    {
+        for (auto _ : state)
+        {
+            for (auto e : c)
+                dummy += e;
+        }
+
+    } else
+    {
+        auto dr = drop_t{}(10);
+        auto ta = take_t{}(10'000);
+        auto v = c | dr | ta | dr | ta | dr | ta;
```

Comment:
That results in a size of 9980, doesn't it?

Is this intended?

Now we are kinda comparing the `void, void` which has to do `1'000'000`  accesses to do one iteration vs. the 9980 accesses the view take/drop combination has to do. (And for `take` we were very particular about also using `1'000'000` elements.

Reply:
good catch